### PR TITLE
Changes in the catalogcd tool to support StepActions in the Catalog.

### PR DESCRIPTION
**Pull request by Senjuti256**
I made changes in the release.go file to support stepActions, apart from adding stepActions to be recognised as a valid catalog resource.
Signed off by: sde@redhat.com (@Senjuti256 )

### CODE REVIEW

**Comment by Senjuti256** [comment]
For validating the generated resources folder you can refer to the Assets of https://github.com/Senjuti256/task-maven/releases/tag/v0.2.4

**Review by vdemeester — APPROVED** [review]

**Comment by vdemeester** [comment]
We need to add support for the `release` to add `StepAction` into `catalog.yaml` and `resources.tar.gz` as well, if not already the case.

**Comment by savitaashture** [comment]
@Senjuti256 can you remove `[.bin/golangci-lint](https://github.com/openshift-pipelines/catalog-cd/pull/86/files#diff-ca1383416d35386c80f0632243837b0a63677fa9184ee0a1f5523f47efd2b69a)`

File: Makefile [hunk 1]
```diff
@@ -90,3 +90,19 @@ pre-commit: ## Run pre-commit hooks script manually
 help:
 	@grep -hE '^[ a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | \
 		awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-17s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: update-go-deps
```
Comment (savitaashture) [inline Makefile:94]: remove this

**Comment by Senjuti256** [comment]
> We need to add support for the `release` to add `StepAction` into `catalog.yaml` and `resources.tar.gz` as well, if not already the case.

@vdemeester those are addressed, in my opinion. If you check the catalog.yaml and resources.tar.gz in the assests of the local release that I have done, you can find them.

**Comment by savitaashture** [comment]
closing as part of this PR https://github.com/openshift-pipelines/catalog-cd/pull/87

**Pull request closed without merging**